### PR TITLE
Fixes #2132 video codes may not be loaded correctly

### LINKED ISSUE #2132: video codes may not be loaded correctly
Yahoo has two different url schemes:

http://{locale}.screen.yahoo.com/{title}-{id}.html
http://{locale}.screen.yahoo.com/{channe...-{id}.html

The first one is loaded correctly, however the second one uses the channel name as video id which results in a broken iframe (or may even result in a wrong video).

Original thread: [video codes may not be loaded correctly](http://community.mybb.com/thread-167116.html)

### PR DESCRIPTION
**Pull request by Stefan-ST**
Fixes #2132

### CODE REVIEW

File: inc/class_parser.php [hunk 1]
```diff
@@ -1352,10 +1352,24 @@ function mycode_parse_video($video, $url)
 				$title = htmlspecialchars_uni($path[3]);
 				break;
 			case "myspacetv":
-				$id = $path[4]; // http://www.myspace.com/video/fds/fds/123
+				if($path[1] == 'video')
+				{
+					$id = $path[4]; // http://www.myspace.com/video/fds/fds/123
```
Comment (Destroy666x) [inline inc/class_parser.php]: This conditional doesn't make any sense.

Comment (Stefan-ST) [inline inc/class_parser.php]: Yeah, I've initially had a different code for the new format. Thanks.

**Comment by PaulBender** [comment]
The Facebook code doesn't work right. $id = $path[4]; needs to be $id = $path[3];

The other two work fine.

**Comment by Stefan-ST** [comment]
@PaulBender What URL have you used? I'm pretty sure I tested this successfully but there might be even more patterns used by Facebook.

**Comment by PaulBender** [comment]
I used this styling:
https://www.facebook.com/viralvidss/videos/377511862456521/

**Comment by Stefan-ST** [comment]
When you e.g. click on the next link you'll see the styling I used. I'll fix that later.

**Comment by Stefan-ST** [comment]
It should both URL formats now.

**Comment by PaulBender** [comment]
Works fine for me

**Pull request merged**